### ISSUE TITLE
MD044 should not flag proper names inside URLs

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#246]
## Bug Description

MD044 (proper-names) is flagging text that appears inside URLs.

## Example

```markdown
[Documentation](https://example.com/getting-started.html)
```

Reports:
```
warning: MD044/proper-names: Proper name 'html' should be capitalized as 'HTML'
```

## Expected Behavior

Text inside URLs (link destinations) should be excluded from proper name checking, as:
1. URLs are case-sensitive and changing them would break links
2. File extensions like `.html` are conventionally lowercase

## Actual Behavior

MD044 flags 'html' in URL paths as needing capitalization to 'HTML'.

## Related

This is similar to how MD044 should exclude code spans and code blocks from checking.
